web/modules/custom/dpl_react_apps/src/Controller/DplReactAppsController.php

```diff
@@ -155,6 +155,16 @@ public function search(): array {
     return $app;
   }
 
+  /**
+   * Get a string with interest periods.
+   *
+   * @return string
+   *   A string with interest periods
+   */
+  public static function getInterestPeriods(): string {
+    return \Drupal::configFactory()->get('dpl_library_agency.general_settings')->get('interest_periods_config');
```

Comment:
I think you should add the general settings as a dependency like the other settings:
https://github.com/danskernesdigitalebibliotek/dpl-cms/blob/0c6d60916b6425243ef1c9b0ce3871b2eab9c4ee/web/modules/custom/dpl_react_apps/src/Controller/DplReactAppsController.php#L44-L48
To keep it persistent and make it easier to test.

Reply:
That is not possible here as the function is static and don't have access to `$this`.

Reply:
Ok I have been digging into the various ways we are handling settings and it is not consistent.
I understand why you need to solve it like this, but I think we can make it more consistent by creating services for the different areas of the configuration (like `DplInstantLoanSettings`) in the future.

Reply:
I hear you but I don't think that this PR should change how the general settings is handled. But "yes" to the idea to change this into the settings pattern now applied in may modules.

Reply:
Todo added and ticket on the way about fixing this into the settings pattern